pkg/controller/upgradeconfig/upgradeconfig_controller.go

```diff
@@ -188,6 +189,29 @@ func (r *ReconcileUpgradeConfig) Reconcile(request reconcile.Request) (reconcile
 			return reconcile.Result{}, err
 		}
 
+		upCoMan, err := upgradeconfigmanager.NewBuilder().NewManager(r.client)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		remoteSpecs, err := upCoMan.GetRemoteSpec()
```

Comment:
Why not use the existing `Refresh()` here? If the current policy is different from the cluster's current UpgradeConfig, the local UpgradeConfig still should to be updated to whatever the new policy is.

`Refresh()` will do that already, and returns a boolean value to indicate if there was indeed a difference in policy vs UpgradeConfig, which also negates the need for the logic further below.

Reply:
Another note - am thinking this should be going in within the `if schedulerResult.IsReady` block, ie. it should only be executed when we've reached the time to upgrade. Where it's placed right now will result in a call/refresh to OCM on each reconcile whilst the UC is in a pending state.